src/components/Location.react.js

```diff
@@ -1,39 +1,85 @@
 import {Component, PropTypes} from 'react';
 /* global window:true */
 
+/**
+ * Update and track the current window.location object through the window.history state
+ */
 export default class Location extends Component {
     constructor(props) {
         super(props);
         this.updateLocation = this.updateLocation.bind(this);
     }
 
     updateLocation(props) {
-        const {pathname, setProps} = props;
+        const {
+            hash,
+            href,
+            pathname,
+            refresh,
+            search,
+            setProps
+        } = props;
 
-        /*
-         * if pathname isn't defined, then it wasn't set by the user
-         * and it needs to be equal to the window.location.pathname
-         * this just happens on page load
+        /**
+         * Check if the field exists in props. If the prop with "fieldName" is not defined,
+         * then it was not set by the user and needs to be equal to the value in window.location.
+         * This only happens on page load (since props will no longer be undefined after componentDidMount).
+         *
+         * @param {string} fieldName
+         *  The name of the prop in window.location and in the component's prop
+         *
+         * @returns {boolean}
+         *  Returns true if the prop with fieldName is different and the window state needs to be updated
          */
-        if (!pathname) {
-            setProps({pathname: window.location.pathname});
-        } else if (pathname !== window.location.pathname) {
-            if (props.refresh) {
-                // Refresh the page
-                window.location.pathname = pathname;
-            } else {
-                // Just push the new location into the URL
-                if (setProps) setProps({pathname});
-                window.history.pushState({}, '', pathname);
+        const checkExistsUpdateWindowLocation = (fieldName) => {
+            const propVal = props[fieldName];
+
+            if (!propVal && window.location[fieldName] && setProps) { // Prop is undefined?
```

Comment:
we use stronger checks for `undefined` here like `R.type(propVal) == 'Undefined'` or even `!R.has(props, propVal)` (http://ramdajs.com/docs/#type, http://ramdajs.com/docs/#has)

Reply:
It's actually a null in some cases (pathname when the page first loads, e.g.), so will update to `(R.type(propVal) === 'Undefined' || propVal === null)` if you prefer. I think that's why the original code had `!propVal`.